### PR TITLE
[TF FE] Refactor translators for Conv2d and Conv3d

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** It allows to convert CNN-Transformer model. Padding was previously incorrect.

**Ticket:** 89061

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: src/frontends/tensorflow/src/utils.cpp [hunk 1]
```diff
@@ -62,3 +66,120 @@ ov::op::PadType ov::frontend::tensorflow::convert_conv_tf_padding(const ov::fron
...
+                                ov::CoordinateDiff& pads_end) {
+    if (spatial_dims_num == 2) {
+        TENSORFLOW_OP_VALIDATION(node,
+                                 tf_explicit_paddings.size() == 8,
+                                 "Conv2D expects 8 padding values for EXPLICIT padding mode.");
+        // prepare pads_begin and pads_end attributes for EXPLICIT padding mode
+        if (is_nhwc) {
+            // For NHWC layout, explicit paddings has the following form:
+            // [0, 0, pad_h1, pad_h2, pad_w1, pad_w2, 0, 0]
+            pads_begin.push_back(tf_explicit_paddings[2]);
```
Comment (itikhono) [inline src/frontends/tensorflow/src/utils.cpp:84]: looks like we need to check that pads_begin is empty here or we can implement like this:

pads_begin = {tf_explicit_paddings[2], tf_explicit_paddings[4]};
pads_end = {tf_explicit_paddings[3], tf_explicit_paddings[5]};

And please do the same below

Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp:84]: I think this is nit.
Now this routine is used from the same function that provides empty `pads_begin`

Comment (itikhono) [reply src/frontends/tensorflow/src/utils.cpp:84]: this function doesn't know it :)
this can lead to mistake

Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp:84]: let me fix this in the next PR due to problems with CI, ok?

Comment (itikhono) [reply src/frontends/tensorflow/src/utils.cpp:84]: you can also implement some lambda-helper:
```
auto fullfill_pads = [&](ov::CoordinateDiff& pads, const std::vector<int64_t>& indexes) {
   pads.resize(indexes.size());
   for (int i = 0; i < indexes.size(); ++i) {
       pads[i] = tf_explicit_paddings[indexes[i]];
   }
};
```
and the code will be like this:
fullfill_pads(pads_begin, {2, 4});

File: src/frontends/tensorflow/src/utils.cpp [hunk 2]
```diff
@@ -62,3 +66,120 @@ ov::op::PadType ov::frontend::tensorflow::convert_conv_tf_padding(const ov::fron
...
+                                 "Conv2D data format is neither NHWC nor NCHW");
+        is_nhwc = (tf_data_format == "NHWC");
+    } else {
+        TENSORFLOW_OP_VALIDATION(node,
+                                 tf_data_format == "NDHWC" || tf_data_format == "NCDHW",
+                                 "Conv3D data format is neither NDHWC nor NCDHW");
+        is_nhwc = (tf_data_format == "NDHWC");
+    }
+
+    // prepare attributes for OpenVINO ConvolutionBackpropData
```
Comment (itikhono) [inline src/frontends/tensorflow/src/utils.cpp:161]: ConvolutionBackpropData?
looks like copy-pasted comment :)

File: src/frontends/tensorflow/src/utils.cpp [hunk 3]
```diff
@@ -62,3 +66,120 @@ ov::op::PadType ov::frontend::tensorflow::convert_conv_tf_padding(const ov::fron
...
+    TENSORFLOW_OP_VALIDATION(node,
+                             spatial_dims_num == 2 || spatial_dims_num == 3,
+                             "Conv2D or Conv3D are supported only.");
+    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() >= 2, "Convolution must have at least two inputs.");
+    auto input = node.get_input(0);
+    auto filter = node.get_input(1);
+
+    // retrieve attributes for Conv2D
+    auto tf_strides = node.get_attribute<std::vector<int64_t>>("strides");
+    auto tf_padding_type = node.get_attribute<std::string>("padding");
```
[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp:134]: Resolved. discussed offline

**Review by itikhono — APPROVED** [review]

File: src/frontends/tensorflow/src/utils.cpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp:161]: nit: Will be fixed in the next PR

**Pull request merged**